core/src/main/java/io/seata/core/logger/StackTraceLogger.java

```diff
@@ -30,10 +32,12 @@
 
     private static final String STACK_TRACE_LOGGER_PREFIX = "[stacktrace]";
 
+    private static final int PERCENT = 100;
+
     public static void info(Logger logger, Throwable cause, String format, Object[] args) {
         if (logger.isInfoEnabled()) {
             int rate = getRate();
-            if (System.currentTimeMillis() % rate == 0) {
+            if (ThreadLocalRandom.current().nextInt(PERCENT / rate) == 0) {
```

Comment:
How to understand rate? If I set the rate to 1000？

Reply:
Look like I misunderstand the mean of rate in the origin logic. I will change it.

Reply:
Fixed. But now the meaning of 
```
client.log.exception
```
was changed.
If set the rate to 1000, It means "show 1 log per 1000 times"